### PR TITLE
add methods for adding metadata and mediafiles to entity

### PR DESCRIPTION
**Pull request by pavelToman**

### CODE REVIEW

File: src/elody/client.py [hunk 1]
```diff
@@ -95,3 +95,25 @@ def upload_file_from_url(self, entity_id, filename, file_url, identifiers=None):
             upload_location, files={"file": mediafile}, headers=self.headers
         )
         return self.__handle_response(response, "Failed to upload mediafile")
+
+    def add_object_metadata(self, collection, identifier, payload):
```
Comment (gverm) [inline src/elody/client.py:99]: Can you sort the functions in this file? (leave private functions with __ or _ on top)

File: src/elody/client.py [hunk 2]
```diff
@@ -95,3 +95,25 @@ def upload_file_from_url(self, entity_id, filename, file_url, identifiers=None):
...
+        if collection == "entities":
+            url = f"{self.elody_collection_url}/{collection}/{identifier}/metadata"
+            payload = payload if isinstance(payload, list) else [payload]
+            response = requests.patch(url, json=payload, headers=self.headers)
+            if (response.status_code == 400) and (
+                response.json()["message"].endswith("has no metadata")
```
Comment (gverm) [inline src/elody/client.py:105]: The parentheses are superfluous. Can you remove them and format the file with Black?

File: src/elody/client.py [hunk 3]
```diff
@@ -95,3 +95,25 @@ def upload_file_from_url(self, entity_id, filename, file_url, identifiers=None):
...
+            ):
+                response = requests.post(url, json=payload, headers=self.headers)
+            return self.__handle_response(response, "Failed to add matadata")
```
Comment (gverm) [inline src/elody/client.py:108]: Typo "matadata" -> "metadata"

File: src/elody/client.py [hunk 4]
```diff
@@ -95,3 +95,25 @@ def upload_file_from_url(self, entity_id, filename, file_url, identifiers=None):
...
+        else:
+            url = f"{self.elody_collection_url}/{collection}/{identifier}"
+            payload = {"metadata": payload if isinstance(payload, list) else [payload]}
+            response = requests.patch(url, json=payload, headers=self.headers)
+            return self.__handle_response(response, "Failed to add matadata")
```
Comment (gverm) [inline src/elody/client.py:113]: Ditto.

File: src/elody/client.py [hunk 5]
```diff
@@ -95,3 +95,25 @@ def upload_file_from_url(self, entity_id, filename, file_url, identifiers=None):
...
+
+    def add_entity_mediafiles(self, identifier,  payload):
```
Comment (gverm) [inline src/elody/client.py:115]: Too many spaces between identifier and payload.

**Review by gverm — CHANGES_REQUESTED** [review]

**Review by gverm — CHANGES_REQUESTED** [review]
Can you sort the functions alphabetically (sorry should have been a bit clearer about that before)?

Also remove the newline between the two imports at the top.

**Comment by gverm** [comment]
@pavelToman Can you squash these commits and make sure to add the authorship info is correct? Make sure that `git log` shows your full name + Inuits email address. Also make sure that your Inuits email address is linked to your GitHub profile.

**Comment by pavelToman** [comment]
I connect my github with an inuits email but the squash should be made by you during the merge I think?

**Pull request merged**